test/websocket/stream.cpp

```diff
@@ -1395,37 +1244,75 @@ class stream_test
-
-                //testInvokable1(ep);
-                testInvokable2(ep);
-                testInvokable3(ep);
-                testInvokable4(ep);
-                //testInvokable5(ep);
-
-                testSyncClient(ep);
-                testAsyncWriteFrame(ep);
-                yield_to_mf(ep, &stream_test::testAsyncClient);
-            }
-            {
-                error_code ec;
-                async_echo_server server{nullptr, 4};
-                server.open(true, any, ec);
-                BEAST_EXPECTS(! ec, ec.message());
-                auto const ep = server.local_endpoint();
-                testSyncClient(ep);
-                testAsyncWriteFrame(ep);
-                yield_to_mf(ep, &stream_test::testAsyncClient);
-            }
+            sync_echo_server server{true, any};
+            auto const ep = server.local_endpoint();
+            //testInvokable1(ep);
+            testInvokable2(ep);
+            testInvokable3(ep);
+            testInvokable4(ep);
+            //testInvokable5(ep);
+            testAsyncWriteFrame(ep);
+        }
+
+        {
+            error_code ec;
+            async_echo_server server{nullptr, 4};
+            server.open(true, any, ec);
+            BEAST_EXPECTS(! ec, ec.message());
+            auto const ep = server.local_endpoint();
+            testAsyncWriteFrame(ep);
         }
+
+        auto const doClientTests =
+            [this, any](permessage_deflate const& pmd)
+            {
+                {
+                    sync_echo_server server{true, any};
+                    auto const ep = server.local_endpoint();
+                    testEndpoint(SyncClient{}, ep, pmd);
+                    yield_to(
+                        [&](yield_context yield)
+                        {
+                            testEndpoint(
+                                AsyncClient{yield}, ep, pmd);
+                        });
+                }
+                {
+                    error_code ec;
+                    async_echo_server server{nullptr, 4};
+                    server.open(true, any, ec);
+                    BEAST_EXPECTS(! ec, ec.message());
+                    auto const ep = server.local_endpoint();
+                    testEndpoint(SyncClient{}, ep, pmd);
+                    yield_to(
+                        [&](yield_context yield)
+                        {
+                            testEndpoint(
+                                AsyncClient{yield}, ep, pmd);
+                        });
+                }
+            };
+
+        permessage_deflate pmd;
+
+        pmd.client_enable = false;
+        doClientTests(pmd);
+
+        pmd.client_enable = true;
+        pmd.client_max_window_bits = 10;
+        pmd.client_no_context_takeover = false;
+        doClientTests(pmd);
+
+        pmd.client_enable = true;
+        pmd.client_max_window_bits = 10;
+        pmd.client_no_context_takeover = true;
+        doClientTests(pmd);
```

Comment:
`server_enable` defaults to false, and I don't see anywhere that it's getting enabled on the server, though I may have missed it. That means that deflate is not being tested by any config.

The coverage report seems to support this hypothesis, because `inflate` and `deflate` are both red. https://codecov.io/gh/vinniefalco/Beast/src/83173f5578294b0e2c0bb03c2a6dd0328a1b2e8c/include/beast/websocket/detail/pmd_extension.hpp#L365...477